mirascope/cli/utils.py

```diff
@@ -33,7 +44,7 @@ def __init__(self):
         self.imports = []
         self.from_imports = []
         self.variables = {}
-        self.classes = []
+        self.classes: list[ClassInfo] = []  # type: ignore
```

Comment:
why type ignore here?

Reply:
mypy throws https://mypy.readthedocs.io/en/latest/error_code_list.html#notify-about-an-annotation-in-an-unchecked-function-annotation-unchecked

i imagine i was supposed to add some arguments in the init but this class doesnt really do that so i opted to add type: ignore instead

Reply:
mypy won't check this because it's inside an untyped function (the init). Because there are no arguments, I think what you should do is add `-> None` to type the init and then also add types to all of the other self variables.

Another interesting possibility would be to double sub-class with `BaseModel` and include all of these attributes as defaulted values? Might specifically require calling the NodeVisitor super init and base model init, so not sure if it's worth it vs. above solution.

Reply:
Actually, another thought: what if you just instantiate each attribute as a class attribute and remove init completely? For example:

```python
class PromptAnalyzer(ast.NodeVisitor):
    """Utility class for analyzing a Mirascope `Prompt` file."""

    imports: list[...] = []
    from_imports: list[...] = []
    variables: dict[..., ...] = {}
    classes: list[ClassInfo] = []
    decorators: list[...] = []
    comments: str = ""

    def visit_import(...) -> None:  # <- this will enable mypy for this function and should be done for all others
        ...
```

Reply:
class attributes caused some issues that instance level did not so i opted to add types to my init instead